ddtrace/contrib/internal/asgi/middleware.py

```diff
@@ -330,3 +516,228 @@ async def wrapped_blocked_send(message):
+            close_span.set_link(
+                trace_id=span.trace_id,
+                span_id=span.span_id,
+                attributes={SPAN_LINK_KIND: SpanLinkKind.RESUMING},
+            )
+            _copy_baggage_and_sampling(close_span, span)
+
+            code = message.get("code")
+            reason = message.get("reason")
+            if code is not None:
+                close_span.set_metric(websocket.CLOSE_CODE, code)
+            if reason:
+                close_span.set_tag(websocket.CLOSE_REASON, reason)
+
+        # Close any receive span at connection close
+        current_receive_span = scope.get("datadog", {}).get("current_receive_span")
+        if current_receive_span:
+            current_receive_span.finish()
+            scope["datadog"].pop("current_receive_span", None)
+
+    def _handle_http_response(
+        self,
+        scope: Mapping[str, Any],
+        message: Mapping[str, Any],
+        span: Span,
+        method: str,
+        response_headers: Optional[Mapping[str, Any]],
+    ):
+        if span and message.get("type") == "http.response.start" and "status" in message:
+            cookies = _parse_response_cookies(response_headers)
+            status_code = message["status"]
+            if self.integration_config.obfuscate_404_resource and status_code == 404:
+                span.resource = " ".join((method, "404"))
+
+            trace_utils.set_http_meta(
+                span,
+                self.integration_config,
+                status_code=status_code,
+                response_headers=response_headers,
+                response_cookies=cookies,
+            )
+            core.dispatch("asgi.start_response", ("asgi",))
+
+    def _handle_websocket_receive_message(
+        self, scope: Mapping[str, Any], message: Mapping[str, Any], recv_span: Span, span: Span, global_root_span: Span
+    ):
+        # Finish the previous receive span before creating a new one
+        previous_receive_span = scope.get("datadog", {}).get("current_receive_span")
+        if previous_receive_span:
+            previous_receive_span.finish()
+            scope["datadog"].pop("current_receive_span", None)
+
+        recv_span.set_tag_str(COMPONENT, self.integration_config.integration_name)
+        recv_span.set_tag_str(SPAN_KIND, SpanKind.CONSUMER)
+        recv_span.set_tag_str(websocket.RECEIVE_DURATION_TYPE, "blocking")
+
+        _set_message_tags_on_span(recv_span, message)
+
+        recv_span.set_metric(websocket.MESSAGE_FRAMES, 1)
+
+        recv_span.set_link(
+            trace_id=span.trace_id,
+            span_id=span.span_id,
+            attributes={SPAN_LINK_KIND: SpanLinkKind.EXECUTED},
+        )
+
+        if self.integration_config.asgi_websocket_messages_inherit_sampling:
+            _set_sampling_inherited(recv_span, global_root_span)
+
+        _copy_baggage_and_sampling(recv_span, span)
+
+        scope["datadog"]["current_receive_span"] = recv_span
```

Comment:
Same here. This can be added to a handler that is called when `asgi.websocker.receive_message` event is triggered.